src/juju/model-selectors.ts

```diff
@@ -1,27 +1,43 @@
 import { createSelector } from "reselect";
 
 import type { ReduxState, ModelsList } from "types";
-import type { ModelInfo, ModelWatcherData } from "./types";
+import type {
+  ApplicationData,
+  MachineData,
+  ModelInfo,
+  ModelWatcherData,
+  RelationData,
+  UnitData,
+} from "./types";
 
 const getModelWatcherData = (state: ReduxState): ModelWatcherData =>
   state.juju.modelWatcherData;
 const getModelList = (state: ReduxState): ModelsList => state.juju.models;
 
-export function getModelInfoByUUID(modelUUID: string) {
+export function getModelWatcherDataByUUID(modelUUID: string) {
+  return createSelector(getModelWatcherData, (modelWatcherData) => {
+    if (modelWatcherData[modelUUID]) {
+      return modelWatcherData[modelUUID];
+    }
+    return null;
```

Comment:
This could be `return modelWatcherData[modelUUID] ?? null` if you want to be fancy.